pkg/cmd/cli/cmd/examples.go

```diff
@@ -0,0 +1,81 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+const examplesLong = `Displays one big list of examples for most commands available in the OpenShift client tools.
+Can be used in conjunction with --output and --template to generate different kinds of output
+formats (json, yaml, templates for generating automatic documentation, etc).`
+
+func NewCmdExamples(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "examples",
+		Short: "Display useful examples for most commands available in the client tools",
+		Long:  examplesLong,
+		Run: func(cmd *cobra.Command, args []string) {
+			outputFormat := kcmdutil.GetFlagString(cmd, "output")
+			outputTemplate := kcmdutil.GetFlagString(cmd, "template")
+
+			p, _, err := kubectl.GetPrinter(outputFormat, outputTemplate)
+			kcmdutil.CheckErr(err)
+
+			examples := extractExamples(cmd.Parent())
+
+			if o := cmd.Flags().Lookup("output"); o != nil && o.Changed {
+				items := []runtime.Object{}
+				for _, example := range examples {
+					items = append(items, example)
+				}
+
+				p.PrintObj(&kapi.List{
+					ListMeta: kapi.ListMeta{},
+					Items:    items,
+				}, out)
+
+				return
+			}
+
+			printExamples(examples, out)
+		},
+	}
+	cmd.Flags().StringP("output", "o", "json", "Output format. One of: json|yaml|template|templatefile.")
+	cmd.Flags().StringP("template", "t", "", "Template string or path to template file to use when -o=template or -o=templatefile.  The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview]")
+	return cmd
+}
+
+func printExamples(examples []*runtime.Unstructured, out io.Writer) {
+	for _, o := range examples {
+		fmt.Printf("\033[1m%v\033[0m\n", o.Object["description"])
```

Comment:
comment to explain what this is doing? I mean, it's completely obvious to me, I'm just thinking of all the people who don't know ANSI escape sequences by heart